docs/source/details/backendconfig.rst

```diff
@@ -21,6 +21,16 @@ Parameters that are directly passed through to an external library and not inter
 The configuration string may refer to the complete ``openPMD::Series`` or may additionally be specified per ``openPMD::Dataset``, passed in the respective constructors.
 This reflects the fact that certain backend-specific parameters may refer to the whole Series (such as storage engines and their parameters) and others refer to actual datasets (such as compression).
 
+A JSON configuration may either be specified as a regular string that can be parsed as a JSON object, or alternatively as a path to a JSON-formatted text file.
+In the latter case, the file path must be prepended by an ampersand ``@``.
+
+.. note::
+
+   Reading text files in parallel openPMD is implemented in MPI-collective manner.
+   The file is read by rank 0 and its contents are then broadcast to all other ranks.
+   This happens at construction of the ``openPMD::Series`` object and when calling ``openPMD::RecordComponent::resetDataset()``, making those calls collective.
+   In order to avoid repeated (collective or non-collective) file access, the file contents for a dataset configuration may be buffered by calling ``openPMD::Dataset::resolveOptions()``.
```

Comment:
I am not sure how useful `resolveOptions` is or if it complicates the API.

In practice, when reading from a file, we don't change the options while a Series is already open. So for that case we can read in and broadcast on `Series()` and have all options available when we need them.

I think that we need additional string-based control on a per-dataset level, but I don't think we need a per-dataset control as a dataset-specific file - I think in practice a per-dataset control would be in the file that the `Series()` call already reads.

Following those assumptions, I think we can simplify this here a bit. Please tell me if discussing this in more detail in a VC can help :)

Reply:
> In practice, when reading from a file, we don't change the options while a Series is already open.

Exactly this is supposed to be the use case. Otherwise, the file would be read upon each `RecordComponent::resetDataset()`. This method allows to read the file once and be done with it.

> I think that we need additional string-based control on a per-dataset level, but I don't think we need a per-dataset control as a dataset-specific file

True somewhat. Major reason I added this is for consistency.

If we allow reading JSON configs from files for datasets, I think some functionality to control MPI-collective operations explicitly like `resolveOptions()` is needed. Otherwise, we can drop it alltogether.

Sure, we can discuss this in a VC :)